src/demo.rs

```diff
@@ -731,22 +752,61 @@ where
                     tracing::info!("\t{}", opt);
                     values.push(opt);
                 }
-
                 self.votes.insert(options.voteidx, values);
+
+                // Replay shunted messages if we have them. This ensures that we don't print VoteCast events for Vote we haven't seen the
+                // VoteOptions event for yet.
+                if !self.shunted_messages.is_empty() {
+                    let mut indices: Vec<usize> = vec![];
+                    for (idx, (vote_idx, shunted_msg)) in
+                        <Vec<(u32, DemoMessage)> as Clone>::clone(&self.shunted_messages)
+                            .into_iter()
+                            .enumerate()
```

Comment:
Wacky syntax, can't you just `for ... in self.shunted_messages.clone().into_iter().enumerate()`? Although it looks like you don't even need an owned copy of this vector so really could it just be `for ... in self.shunted_messages.iter().enumerate()` and get borrows?

Reply:
Actually may possibly be redundant, see later messages.